src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportTest.java

```diff
@@ -0,0 +1,231 @@
+
+         //List<RespondentSumTypeItem> respondentCollection = new ArrayList<>();
+         var respondentSumTypeItem = new RespondentSumTypeItem();
+         var respondentSumType = new RespondentSumType();
+         var firstRespondentAddress = new Address();
+         firstRespondentAddress.setPostCode("DH1 1AE");
+         respondentSumType.setRespondentAddress(firstRespondentAddress);
+
+         var firstRespondentET3Address = new Address();
+         firstRespondentET3Address.setPostCode("DH1 1AJ");
+         respondentSumType.setResponseRespondentAddress(firstRespondentET3Address);
+         respondentSumTypeItem.setValue(respondentSumType);
+         respondentCollection.add(respondentSumTypeItem);
+
+         var respondentSumTypeItemTwo = new RespondentSumTypeItem();
+         var respondentSumTypeTwo = new RespondentSumType();
+         var secondRespondentAddress = new Address();
+         secondRespondentAddress.setPostCode("DH5 9AJ");
+         respondentSumTypeTwo.setRespondentAddress(secondRespondentAddress);
+         respondentSumTypeItemTwo.setValue(respondentSumTypeTwo);
+         respondentCollection.add(respondentSumTypeItemTwo);
+
+         var claimantAddressUK = new Address();
+         claimantAddressUK.setPostCode("DH3 8HL");
+         var claimant = new ClaimantType();
+         claimant.setClaimantAddressUK(claimantAddressUK);
+
+         var submitEventOne = caseDataBuilder
+             .withEthosCaseReference("18000012/2022")
+             .withReceiptDate("2021-12-14")
+             .withClaimantType(claimant)
+             .withClaimantWorkAddressType(null)
+             .withRespondentCollection(respondentCollection)
+             .buildAsSubmitEvent(ACCEPTED_STATE);
+         submitEvents.add(submitEventOne);
+         when(claimsByHearingVenueReportDataSource.getData(
+             UtilHelper.getListingCaseTypeId(LEEDS_LISTING_CASE_TYPE_ID), START_DATE, END_DATE))
+             .thenReturn(submitEvents);
+
+         var expectedReportDetailEntriesCount = submitEvents.size();
+         var expectedRespondentPostCode = firstRespondentAddress.getPostCode();
+         var expectedFirstRespondentET3AddressPostCode = firstRespondentET3Address.getPostCode();
+         var listingDetails = new ListingDetails();
+         var reportData = claimsByHearingVenueReport
+             .generateReport(LEEDS_LISTING_CASE_TYPE_ID, null);
+         var actualRespondentPostCode = reportData.getReportDetails().get(0).getRespondentPostcode();
+         var actualRespondentET3PostCode = reportData.getReportDetails().get(0).getRespondentET3Postcode();
+
+         assertEquals(expectedReportDetailEntriesCount, reportData.getReportDetails().size());
+         assertEquals(expectedRespondentPostCode, actualRespondentPostCode);
+         assertEquals(expectedFirstRespondentET3AddressPostCode, actualRespondentET3PostCode);
+     }
+
+    @Test
+    void shouldShowReportDetailEntriesSortedByEthosCaseReference() {
```

Comment:
May be a test when ClaimantAddress is not null but postcode is null?

Reply:
Ok. I have a look.

Reply:
New test added.